### ISSUE TITLE
Replace environment.py with Unittests with mocking

### ISSUE THREAD

**Issue by GREER-J** [GREER-J/NU_Marine#2]
A/A

Currently one key requirement of the project is that it is 'seen to work' i.e we can perform tests to ensure that the final product will behave as desired.

Currently this is implemented by creating a class 'environment' kept in 'environment.py' that handles interactions between the state machine and the 'environment' simulating responses. One large downside of this is the state machine cannot work with environment.py and run it's normal production code. Leading to a situation where the 'final' code will have to be replaced at the tail end of development with code that 'should do the same thing'. However this funcitonality would not be able to be verified with the current setup.

It would be better if this was instead done through Unittesting, with integrated mocking of key responses.

TLDR:
 - Replace environment.py with unittests with mocking of key responses.
ADV:
 - Cleaner code
 - Ability to have test production code
 - Makes future development easier by verifying that desired behaviour is maintained
 DIS:
 - Less readable to non-software engineers

**Comment by GREER-J** [comment]
Mock Pyserial response:

https://stackoverflow.com/questions/46492209/how-to-emulate-data-from-a-serial-port-using-python-3-unittest-mocks
